azure-fluid-relay-preview-pr/articles/howtos/hosting.md

```diff
@@ -0,0 +1,258 @@
+
+1. `InsecureTokenProvider`: This type of implementation is only recommended for development purposes as it exposes the tenant key secret in your client-side code bundle. You can add the below `config` in [app.ts](https://github.com/microsoft/FluidHelloWorld/blob/main/src/app.ts).
+
+```typescript
+import { AzureClient, AzureConnectionConfig } from "@fluidframework/azure-client";
+
+const config: AzureConnectionConfig = {
+    tenantId: "myTenantId",
+    tokenProvider: new InsecureTokenProvider("myTenantKey", { id: "123", name: "Test User" }),
+    orderer: "https://myOrdererUrl",
+    storage: "https://myStorageUrl",
+};
+const client = new AzureClient(config);
+```
+
+2. `AzureFunctionTokenProvider`: This implementation can be used for fetching the token from your own backend service that is responsible for signing it with the tenant key. It provides you with a secured way to generate token and pass the generated token back to the client app. `AzureFunctionTokenProvider` is an implementation that fulfills the `ITokenProvider` interface without exposing the tenant key secret in client-side code. Developers would require their own backend `HTTPTrigger Azure Function` for fetching the token, thereby passing the url to `AzureFunctionTokenProvider` in [app.ts](https://github.com/microsoft/FluidHelloWorld/blob/main/src/app.ts).
+
+```typescript
+import { AzureClient, AzureConnectionConfig } from "@fluidframework/azure-client";
+
+const config: AzureConnectionConfig = {
+    tenantId: "myTenantId",
+    tokenProvider: new AzureFunctionTokenProvider("myAzureFunctionUrl"+"/api/GetFrsToken", { userId: "test-user",userName: "Test User" }),
+    orderer: "https://myOrdererUrl",
+    storage: "https://myStorageUrl",
+};
+const client = new AzureClient(config);
+```
+
+<br />
+
+> [!NOTE]
+> The below implementation can be used for `Static Web Apps` only.
+
+3. `Static Web Apps` allow you to develop a full-stack web site without needing to deal with the server-side configuration of an entire web hosting environment and not exposing the token on the client-side bundle code. You can fetch the token by setting up your own backend `api`. Developers can create their own `HTTPTrigger Azure Function` under an `api` folder in the `FluidHelloWorld` repository itself. Below, `GetFrsToken` is an `HTTPTrigger Azure Function` which is responsible for fetching token from your very own backend. 
+
+<img src='./resources/images/apiFileExplorer.png' style="height: 300px"> 
+
+<br />
+<br />
+
+The below code snippet can help you with creating `HTTPTrigger` function for fetching the token by passing in your tenant key.
+
+```typescript
+import { AzureFunction, Context, HttpRequest } from "@azure/functions";
+import { ScopeType } from "@fluidframework/protocol-definitions";
+import { generateToken, generateUser } from "@fluidframework/server-services-utils";
+
+const key = "myTenantKey";
```

Comment:
Let's be explicit that this line is the one that they would need to replace with their own key string